### PR TITLE
optimized taunts

### PR DESCRIPTION
**Pull request by Idunno1**

### CODE REVIEW

File: scripts/main/battle_taunt.lua [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+function Mod:updateBattleTaunt()
+    local toque_equipped = false
+    for _,party in ipairs(Game.party) do
+        if party:checkArmor("pizza_toque") then toque_equipped = true end
+    end
+    if
+        (toque_equipped or Game.save_name:upper() == "PEPPINO" or self.let_me_taunt)
+        and Input.pressed("v", false)
+        and self.taunt_cooldown == 0
+        and (Game.state == "BATTLE" and Game.world.state == "GAMEPLAY" and not Game.world:hasCutscene())
```
Comment (Dobby233Liu) [inline scripts/main/battle_taunt.lua:26]: TBH checking Game.world.state and Game.world:hasCutscene is pretty pointless here

File: scripts/main/battle_taunt.lua [hunk 2]
```diff
@@ -0,0 +1,60 @@
+function Mod:initBattleTaunt()
+    self.taunt_cooldown = 0
+    self.state_blacklist = {
+        "DEFENDINGBEGIN",
+        "DEFENDING",
```
Comment (Dobby233Liu) [inline scripts/main/battle_taunt.lua:5]: IMO it would be nice if you note that taunting is handled by our Soul hook in DEFENDING

Comment (Dobby233Liu) [reply scripts/main/battle_taunt.lua:5]: Also I would like to remind that this is still in the Mod namespace so I suggest to prefix the variable name

File: scripts/main/battle_taunt.lua (hunk 1 above)
Comment (Idunno1) [reply scripts/main/battle_taunt.lua:26]: forgot to remove those, whoops

File: scripts/main/battle_taunt.lua (hunk 2 above)
Comment (Idunno1) [reply scripts/main/battle_taunt.lua:5]: i was thinking about it after the pr was merged, whoops

**Pull request merged**